.github/workflows/codeql.yml

```
@@ -0,0 +1,60 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", "dev" ]
+  pull_request:
+    branches: [ "master", "dev" ]
+  schedule:
+    - cron: '16 19 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - SDK: "$NANOS_SDK"
+            artifact: hathor-app-nanoS
+          - SDK: "$NANOX_SDK"
+            artifact: hathor-app-nanoX
+          # - SDK: "$NANOSP_SDK"
+          #   artifact: hathor-app-nanoSP
+        language: [ 'c-cpp' ]
+    container:
+      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-legacy:latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
```

suggestion: Specify a commit hash to avoid supply chain issues:
```suggestion
      # https://github.com/actions/checkout/releases/tag/v4.0.0
      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
```

---

I think this is valid, but since this is a github security action I think we should always keep the latest version running.